docs/ide-integrations/visual-studio/Visual-Studio-Integration-Editing-Features.md

```diff
@@ -0,0 +1,137 @@
+**Display items** in the list begin with \"Reqnroll\".
+
+In addition to highlighting keywords, comments, tags etc., unbound steps
+and parameters in feature files are highlighted when editing the file in
+Visual Studio. The following syntax highlighting is used by default:
+
+-   Purple: unbound steps
+-   Black: bound steps
+-   Red: parameters in bound steps
+
+You can thus tell immediately which steps in a feature file have been
+bound.
+
+```{note}
+In Reqnroll Visual Studio 2022 plugin a project must be **built** for
+syntax highlighting to update.
+```
+
+## IntelliSense (auto-completion) for Keywords and Steps
+
+IntelliSense makes Reqnroll easy to use when integrated with Visual
+Studio. IntelliSense uses find-as-you-type to restrict the list of
+suggested entries.
+
+### Gherkin Files
+
+IntelliSense is available in feature files for the following:
+
+-   Gherkin keywords (e.g. `Scenario`, `Given` etc.)
+-   Existing steps are listed after a `Given`, `When` or `Then`
+    statement, providing quick access to your current steps definitions.
+
+```{note}
+Note that all the steps in all \"\*.feature\" files\" that match the
+current type (Given, When, Then) are displayed
+```
+
+![image](../../_static/images/vs2022inteli.png)
+
+
+### Code Files
+
+IntelliSense is also available for the Gherkin keywords in your code
+files.
+
+### IntelliSense settings
+
+You must edit the [reqnroll.json](../../installation/configuration.md) config file access other settings as per below:
+
+![IntelliSense 2022](../../_static/images/inteli2022.png)
+
+## Outlining and Comments in Feature Files
+
+Most of the items in the **Edit** menu work well with Reqnroll feature
+files, for example:
+
+-   You can comment and uncomment selected lines (\'#\' character) with
+    the default shortcut for comments (Ctrl+K Ctrl+C/Ctrl+K Ctrl+U) or
+    from the menu
+-   You can use the options in the **Edit \| Outlining** menu to expand
+    and contract sections of your feature files ![VS2010 Edit
+    menu](../../_static/images/outlining_editor.png)
+
+## Table Formatting
+
+Tables in Reqnroll are also expanded and formatted automatically as you
+enter column names and values:
+
+```{figure} /_static/images/tablegifvs.gif
+:alt: Formatted table
+
+Formatted table
+```
+
+## Document Formatting
+
+Document formatting is also available. It automatically re-indents code
+and fixes blank lines, comments, etc.
+
+You can find this option under Edit-\>Advanced-\>Format document or
+alternatively use the Ctrl+K, Ctrl+D shortcut:
+
+```{figure} /_static/images/format-doc.png
+:alt: Format document
+
+Format document
+```
+
+Below is a feature file document which is not indented correctly:
+
+```{figure} /_static/images/format-doc-before.png
+:alt: Unformatted document
+
+Unformatted document
+```
+
+After the `Format Document` command:
+
+```{figure} /_static/images/format-doc-after.png
+:alt: Formatted document
+
+Formatted document
+```
+
```

Comment:
Probably would make sense to mention here that the formatting options (indentations, etc.) of Gherkin can be configured.